pkg/onchain/datastore.go

```diff
@@ -54,8 +55,71 @@ func populateDatastore(contract ContractDatastore) ([]byte, error) {
 	return serializedDatastore, nil
 }
 
+// getContractByteCodeKey returns the key for the deployed contract bytecode in the datastore.
+func getContractByteCodeKey() []byte {
+	return convert.U64ToBytes(contractByteCodeKey)
+}
+
+// getArgsKey returns the key for the deployed contract constructor parameters in the datastore.
+func getArgsKey() []byte {
+	lengthPrefix := convert.U32ToBytes(byteArrayLengthPrefix)
+	argsKeySuffix := []byte{0}
+
+	tempKey := append(getContractByteCodeKey(), lengthPrefix...)
+
+	return append(tempKey, argsKeySuffix...)
+}
+
+// getCoinsKey returns the key in the datastore for the amount of MAS to be sent to the deployed contract.
+func getCoinsKey() []byte {
+	lengthPrefix := convert.U32ToBytes(byteArrayLengthPrefix)
+	coinsKeySuffix := []byte{1}
+
+	tempKey := append(getContractByteCodeKey(), lengthPrefix...)
+
+	return append(tempKey, coinsKeySuffix...)
+}
+
+/*
+DatastoreToDeployedContract If the datastore is a valid datastore for a deployed contract,
+it will return the contract's bytecode, args and coins
+If the datastore is not a valid datastore for a deployed contract,
+it will return an empty ContractDatastore and success = false.
+*/
+func DatastoreToDeployedContract(datastore []DatastoreEntry) (contractDatastore ContractDatastore, success bool) {
+	if !isContractDeployDatastore(datastore) {
+		return ContractDatastore{}, false
+	}
+
+	var contract ContractDatastore
+	contract.ByteCode = datastore[1].Value
+	contract.Args = datastore[2].Value
+
+	var err error
+
+	contract.Coins, err = convert.BytesToU64(datastore[3].Value)
+	if err != nil {
+		return ContractDatastore{}, false
+	}
+
+	return contract, true
+}
```

Comment:
What we usually to is return  data, error  for function return value.
When error = nil its equivalent to your success=true
when error != nil we can get the error message.

what do you think?

Reply:
then in the calling function we can check the error to see whats to do